lib/puppet-syntax/hiera.rb

```diff
@@ -42,30 +42,28 @@ def check_eyaml_data(name, val)
     end
 
     def check_eyaml_blob(val)
-      return unless val =~ /^ENC\[/
-
-      val.sub!('ENC[', '')
+      # strip newlines and extra spaces
+      val.gsub!(/\n/, '')
       val.gsub!(/\s+/, '')
-      if val !~ /\]$/
-        return "has unterminated eyaml value"
-      else
-        val.sub!(/\]$/, '')
-        method, base64 = val.split(/,/)
-        if base64 == nil
-          base64 = method
-          method = 'PKCS7'
-        end
+      
+      encodes = val.scan(/ENC\[.*?\]/)
```

Comment:
Why `.*?` as a regex? The `?` doesn't make sense to me. Also wondering why you don't drop the ENC part via groups:
```
[1] pry(main)> 'ENC[PKCS7,aGVsbG8sf&IHdvcmxk==]'.scan(/ENC\[(.+)\]/)
=> [["PKCS7,aGVsbG8sf&IHdvcmxk=="]]

```

```suggestion
      encodes = val.scan(/ENC\[(.*)\]/)
```

You can even already do the splitting here:
```suggestion
      encodes = val.scan(/ENC\[([^,]+),?(.+)?\]/)
```

Note I made the last part optional so you can still detect an invalid format.

Reply:
https://github.com/voxpupuli/hiera-eyaml/blob/master/lib/hiera/backend/eyaml_backend.rb#L101-L103

The `?` didn't really make sense to me either but it seemed appropriate since that's what hiera-eyaml uses.

Reply:
I'm all for improving it and my preference would be to do a scan once but I didn't find something that works

`encodes = val.scan(/ENC\[([^,]+),?(.+)?\]/)`

This fails the tests. Take the example `ENC[KMS,aGVsbG8sIdGHdvcmxk==]ENC[KMS,aGVsbG8sIdGHdvcmxk==]`

Gives us these match groups.
```
1. | KMS
2. | aGVsbG8sIdGHdvcmxk==]ENC[KMS,aGVsbG8sIdGHdvcmxk==
```

Reply:
This works for me:
```ruby
data = 'ENC[KMS,aGVsbG8sIdGHdvcmxk==]ENC[KMS,aGVsbG8sIdGHdvcmxk==]'
data.scan(/ENC\[([^,]+),?([^\]]+)?\]/)
```

Returns `[["KMS", "aGVsbG8sIdGHdvcmxk=="], ["KMS", "aGVsbG8sIdGHdvcmxk=="]]`

Reply:
Hey @wershlak @ekohl 

Following-up your conversation about the question mark `?` in the REGEX, it seems that we can remove it right from the upstream project, here is the PR to the change:

https://github.com/voxpupuli/hiera-eyaml/pull/313

Thanks!